src/PitchDetection/index.js

```diff
@@ -16,12 +16,12 @@ class PitchDetection {
     this.modelName = modelName;
     this.audioContext = audioContext;
     this.stream = stream;
-    this.loadModel();
+    this.ready = this.loadModel().then(() => this);
```

Comment:
I'd also posit that this is a bugfix; pitchdetection has an async loading step but no callback.

Reply:
👍

Reply:
Note that this doesn't actually add a callback to the constructor.